### PR TITLE
RealmConfiguration.path should report the full Realm path.

### PR DESCRIPTION
**Pull request by nhachicha**

### CODE REVIEW

**Review by cmelchior — COMMENTED** [review]
While the changes here are good and we should probably merge them, I don't think this PR solves the underlying issue.

If people are using `.` in their own paths, we would still not print the absolute path.

I was thinking more that we should use something like https://docs.oracle.com/javase/7/docs/api/java/io/File.html#getAbsolutePath() for Java and perhaps https://stackoverflow.com/questions/17395556/how-would-i-resolve-a-relative-path-in-objective-c for Cocoa. Unfortunately Cocoa does seem to have some issues resolving relative paths, so not sure how much we can achieve here. One solution might be to default `currentDirectoryPath` as root for any relative path.

File: packages/library-base/src/jvmMain/kotlin/io/realm/internal/platform/SystemUtilsJvm.kt [hunk 1]
```diff
@@ -7,7 +7,7 @@ actual val OS_NAME: String = System.getProperty("os.name")
 actual val OS_VERSION: String = System.getProperty("os.version")
 
 @Suppress("FunctionOnlyReturningConstant")
-actual fun appFilesDirectory(): String = "."
+actual fun appFilesDirectory(): String = System.getProperty("user.dir") ?: "."
```
[the remark this answers is not in the document]
Comment (nhachicha) [reply packages/library-base/src/jvmMain/kotlin/io/realm/internal/platform/SystemUtilsJvm.kt:10]: This is still using the current directory, the only change here is that it gets the dir as a full path

File: packages/library-base/src/macosMain/kotlin/io/realm/internal/platform/SystemUtils.kt [hunk 2]
```diff
@@ -2,5 +2,5 @@ package io.realm.internal.platform
 
 @Suppress("FunctionOnlyReturningConstant")
 actual fun appFilesDirectory(): String {
-    return "."
+    return platform.Foundation.NSFileManager.defaultManager.currentDirectoryPath
```
[the remark this answers is not in the document]
Comment (nhachicha) [reply packages/library-base/src/macosMain/kotlin/io/realm/internal/platform/SystemUtils.kt:5]: ditto

**Comment by rorbech** [comment]
For some reason I cannot review this 🤷, but I think there is an issue with only expanding `./` into `appFilesDirectory`. We should prepend the app directory for all relative paths and not only the ones starting with `./`.

**Pull request merged**